adapters/orbidder/orbidder.go

```diff
@@ -140,6 +140,23 @@ func (rcv OrbidderAdapter) MakeBids(_ *openrtb2.BidRequest, _ *adapters.RequestD
 	return bidResponse, nil
 }
 
+func getMediaTypeForImp(impID string, imps []openrtb2.Imp) openrtb_ext.BidType {
+	for _, imp := range imps {
+		if imp.ID == impID {
+			if imp.Banner != nil {
+				return openrtb_ext.BidTypeBanner
+			} else if imp.Video != nil {
+				return openrtb_ext.BidTypeVideo
+			} else if imp.Native != nil {
+				return openrtb_ext.BidTypeNative
+			} else if imp.Audio != nil {
+				return openrtb_ext.BidTypeAudio
+			}
+		}
+	}
+	return openrtb_ext.BidTypeBanner
+}
```

Comment:
Warning: this is an anti-pattern. This code assumes that if there is multi-format request then mediatype is defaulting to banner. 
Ideally, Prebid server expects media type set in adapter response. we recommend implementing a pattern where adapter server sets the [MType](https://github.com/prebid/openrtb/blob/53ac1b9ca7ddf860988ff69bf5018b32ced1fbc5/openrtb2/bid.go#L334) field in the response to accurately determine the media type for the impression.

Reply:
@onkarvhanumante thanks a lot for the hint. we will change the implementation like you describe.
We did look at other adapters, and found a lot of this solutions.
But like you explain it is much more straightforward to set the type at the bid response.

Reply:
@onkarvhanumante, thanks again four you input. i did some changes.
with adding a multibid testcase i found as well an error at our adapter.

Commit: https://github.com/prebid/prebid-server/pull/2924/commits/d605e8c433f514e304920cfbc8574512b0280943